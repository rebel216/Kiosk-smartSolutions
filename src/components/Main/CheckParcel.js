import React, { Component, useState } from 'react'
import { Link } from "react-router-dom";
import { w3cwebsocket as W3CWebSocket } from 'websocket';
import { useEffect } from 'react';
import { useNavigate } from 'react-router-dom';
const {WDS_SOCKET_PORT2} = process.env
const client = new W3CWebSocket("ws://localhost:9000");
export default function CheckParcel () {


  //const weight1 = JSON.parse(sessionStorage.weight1);

  const [weight, setWeight] = useState("");




    client.onmessage = (e) => {
      const newObj = e.data;
      setWeight(newObj);


    };

    setTimeout(() => {
      client.close();

    }, "5000")

    client.onopen = () => {
      console.log("Connected");
    };



    client.onclose = () => {
      console.log("Closed...");
    };
  const w1 = parseFloat("12.12sa")//(weight1.substring(0,6))
  const w = parseFloat("12.12sa")//(weight.substring(0, 6))
  
  const checkWeight = () => {
    console.log("clicked")
    client.onmessage = (e) => {
      const newObj = e.data;
      setWeight(newObj);


    };

    setTimeout(() => {
      client.close();

    }, "5000")

    client.onopen = () => {
      console.log("Connected");
    };



    client.onclose = () => {
      console.log("Closed...");
    };
 }

  return (
    <div className='heading'>
        <h1>Place the parcel inside the Machine Again</h1>
        <div className='login-box'>
      


        
        {w==w1  ? (
        <Link  to="/payment" className="link1"><button className="btn btn-primary w-75 mb-2">Make PayMent</button></Link>
      ) : (
              <><h1>Weight missmatch. Please Place the same parcel</h1>
                <button onClick={checkWeight} className="error">Click to Check Again</button></>
      )}
       
</div>

      </div>
    )

}


