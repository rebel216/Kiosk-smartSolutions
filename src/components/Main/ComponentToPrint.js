import React from "react";




const ComponentToPrint = () => {
    var user = JSON.parse(sessionStorage.user);
    return (
        <div className="main-cointainer">
            <h2>Compnent2</h2>

<p>{user.fname} </p>

        </div>
    )
}

export default ComponentToPrint;
