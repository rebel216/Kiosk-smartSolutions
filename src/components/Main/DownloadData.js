import React from "react";
import axios from "axios";
import BASE_URL from "../../api";
import * as FileSaver from "file-saver";
import * as XLSX from "xlsx";

function DownloadData() {
  const fetchData = async () => {
    const response = axios.get(
      `${BASE_URL}api/v1/parcels/${
        JSON.parse(localStorage.getItem("user")).id
      }`,
      {
        headers: {
          Authorization: JSON.parse(localStorage.getItem("user")).token,
        },
      }
    );
    const data1 = (await response).data;

    const fileType =
      "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet;charset=UTF-8";
    const fileExtension = ".xlsx";
    console.log(data1);

    const ws = XLSX.utils.json_to_sheet(data1);
    const wb = { Sheets: { data: ws }, SheetNames: ["data"] };
    const excelBuffer = XLSX.write(wb, { bookType: "xlsx", type: "array" });
    const data = new Blob([excelBuffer], { type: fileType });
    FileSaver.saveAs(data, "data" + fileExtension);
  };

  return (
    <div className="login-box">
      <div>
        <h1>Send Data to registered email.</h1>
      </div>
      <div>
        <div className="form-group mt-3 d-flex flex-column  align-items-center justify-content-center">
               
        <button className="btn btn-primary w-75 mb-2" onClick={fetchData}>
          Send Email
        </button></div>
        <br></br>
      </div>
    </div>
  );
}

export default DownloadData;
