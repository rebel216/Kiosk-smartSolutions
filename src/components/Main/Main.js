import React, { useEffect } from 'react';
import { NavLink } from 'react-router-dom';
import '../assets/styles/App.css';

const Main = () => {
  useEffect(() => {

  }, []);

  return (
      <>

      <div className="ticket">

<p>
                  <NavLink to="/RegisteredPost" className='main-link'>
                  RegisteredPost
          </NavLink> </p>


          <p>            <NavLink to="/SpeedPost" className='main-link'>
                  SpeedPost
          </NavLink>
          </p>


      </div>
    </>
  );
};

export default Main;
