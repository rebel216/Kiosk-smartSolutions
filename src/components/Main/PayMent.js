//import React, { useState } from 'react';
//import { QRCode } from 'react-qrcode-logo';
import React, { useState } from 'react'
import axios from 'axios'
import { useNavigate } from 'react-router-dom'



function PayMent() {

    const navigate = useNavigate();
function loadScript(src) {
	return new Promise((resolve) => {
		const script = document.createElement('script')
		script.src = src
		script.onload = () => {
			resolve(true)
		}
		script.onerror = () => {
			resolve(false)
		}
		document.body.appendChild(script)
	})
}

    const options = {
        key: "rzp_test_c5Tw7vHbkg3thN", // Enter the Key ID generated from the Dashboard
        amount: "10000", // caculate amnt later 
        currency: "INR",
    }
    
    async function displayRazorpay() {
        const res = await loadScript(
            "https://checkout.razorpay.com/v1/checkout.js"
        );

        if (!res) {
            alert("Razorpay SDK failed to load. Are you online?");
            return;
        }
      const result = await axios.post("http://localhost:9000/payment/orders");
    //     console.log(result)
    //     // creating a new order
    //  console.log("clicked")

        if (!result) {
            alert("Server error. Are you online?");
            return;
        }

       // Getting the order details back
       const { amount, id: order_id, currency } = result.data;
       const options = {
            key: "rzp_test_c5Tw7vHbkg3thN", // Enter the Key ID generated from the Dashboard
            amount: amount.toString(),
            currency: currency,
            name: "Xoqovo Corp",
            description: "Test Transaction",
            
            order_id: order_id,
            handler: async function (response) {
                const data = {
                    orderCreationId: order_id,
                    razorpayPaymentId: response.razorpay_payment_id,
                    razorpayOrderId: response.razorpay_order_id,
                    razorpaySignature: response.razorpay_signature,
                };

               // const result = await axios.post("http://localhost:9000/payment/success",data);
       navigate("/success")
                alert("PAyment succesfull");
            },
            prefill: {
                name: "Xoqovo",
                email: "Xoqovo@example.com",
                contact: "1231231231",
            },
            notes: {
                address: "Xoqovo Corporate Office",
           },
          
            theme: {
                color: "#61dafb",
            },
        };
      
        
        const paymentObject = new window.Razorpay(options);
        paymentObject.open();
}
	

	return (
		  <div className="login-box">
            <header className="App-header">
                
                <h2>Make payment </h2>
                <div className="form-group mt-3 d-flex flex-column  align-items-center justify-content-center">
                <button className="btn btn-primary w-75 mb-2" onClick={displayRazorpay}>
                    Pay â‚¹100
                </button></div>
            </header>
        </div>
	)
}

export default PayMent



// import qr from '../assets/images/qr.PNG'
// import { Link } from 'react-router-dom';

// export default function PayMent() {
//   var user = JSON.parse(sessionStorage.user);
//   const money = user.user.price
//   console.log(money)
//   return (
//     <div className='login-box'>
// <img src={qr} alt="QR" ></img>

//       {/* <QRCode
//         value={`upi://pay?pa=intellemo@icici&pn=Intellemo&tn=cftrhwetaw4gta&am=${money}`} //generate your own Qr code.
//         size="300"

//       /> */}
//       <p>Scan the code using PhonePe, Google Pay or Paytm</p>
//       < Link to="/recieptprint" className="link">Make Payment</Link>
//     </div>
//   );
// }
