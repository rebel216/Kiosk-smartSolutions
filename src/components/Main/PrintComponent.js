import React, { useRef } from "react";
import Barcode from 'react-barcode';
import { Navigate, useNavigate } from "react-router-dom";
import logo from '../assets/images/logo.png'

export default function PrintComponent() {
  const navigate = useNavigate();
  const time = new Date().toLocaleString()
  const handleClick = () => {
        
  

   window.print()
    navigate("/ScanBarcode")
  }
  var user = JSON.parse(sessionStorage.user);
  var barcode2 = JSON.parse(sessionStorage.barcode2);

  console.log(user.user)
  return (
    <>
    
      <div>
        
      <div className="login-box">

  <Barcode value={barcode2} />
        
          

          <h6 className="centered">Date:<span>  </span> {time} </h6>
          <h6 className="centered">Reciever:</h6>
          <h6 className="centered"> {user.user.rfname}<span>  </span>{user.user.rlname}</h6>

          <h6 className="centered">Address:<span>  </span>{user.user.address2}</h6>
          <h6 className="centered">  State:<span>  </span>{user.user.state2}</h6>
          <h6 className="centered">  City:<span>  </span>{user.user.city2}</h6>
          <h6 className="centered"> Mobile:<span>  </span>{user.user.phone2}</h6>
          <h6 className="centered">  PinCode:<span>  </span>{user.user.pincode2}</h6>
          <h6 className="centered">Sender:</h6>
          <h6 className="centered"> {user.user.fname}<span>  </span>{user.user.lname}</h6>

          <h6 className="centered">Address:<span>  </span>{user.user.address1}</h6>
          <h6 className="centered">  State:<span>  </span>{user.user.state1}</h6>
          <h6 className="centered">  City:<span>  </span>{user.user.city1}</h6>
          <h6 className="centered"> Mobile:<span>  </span>{user.user.phone1}</h6>
          <h6 className="centered">  PinCode:<span>  </span>{user.user.pincode1}</h6>
      
        </div> <div className="">
        <button className='link1' onClick={handleClick}>PRINT</button>
        </div>
       </div>
    </>
  )
}
