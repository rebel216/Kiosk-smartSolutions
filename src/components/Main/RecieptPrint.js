import React, { Component } from 'react'
import { useNavigate } from 'react-router-dom';
import { Link } from 'react-router-dom';
import logo from '../assets/images/logo.png'

// move the motor here.

const RecieptPrint = () => {

    const navigate = useNavigate();
    const today = new Date(),
    time =  new Date().toLocaleString()
  const handleClick = () => {
    window.print()
    navigate("/success")
  }
  var user = JSON.parse(sessionStorage.user);


    return (
        <>

      <div className="login-box">

      <img src={logo} alt="Logo" className="logo"/>

                <h3 className="centered"> Name: {user.user.fname}<span>  </span>{user.user.lname}</h3>
                <h3 className="centered">Date:<span>  </span> {time} </h3>
                <h3 className="centered">Tracking No:<span>  </span>{user.user.barcode}</h3>
                <h3 className="centered">Weight:<span>  </span>{user.user.weight1}</h3>
                <h3 className="centered">  From :<span>  </span>{user.user.state1}</h3>
                <h3 className="centered">  To:<span>  </span>{user.user.state2}</h3>
                <h3 className="centered"> Mobile:<span>  </span>{user.user.phone1}</h3>
                <h3 className="centered">  Paid: Rs.<span>  </span>{user.user.price}</h3>


            <button className='main-button' onClick={handleClick}>PRINT</button>
      </div>
      </>
    )

}

export default RecieptPrint
