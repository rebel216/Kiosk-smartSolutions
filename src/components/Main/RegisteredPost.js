import React from 'react';
import { useState } from 'react';
import { useDispatch } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import '../assets/styles/App.css';
import Pincode from "react-pincode";
import { w3cwebsocket as W3CWebSocket } from 'websocket';
const client = new W3CWebSocket("ws://127.0.0.1:7000");

export default function RegisteredPost (){
  const [weight, setWeight] = useState("");

  client.onmessage = (e) => {
    const newObj = e.data;
    setWeight(newObj);
    sessionStorage.weight1 = JSON.stringify(weight);
    console.log(newObj);

  };

  setTimeout(() => {
    client.close();

  }, "5000")

  client.onopen = () => {
    console.log("Connected");
  };



  client.onclose = () => {
    console.log("Closed...");
  };

  //return () => client.OPEN && client.close();

  const phone = "234234234"//JSON.parse(sessionStorage.phone);

  const barcode2 = "RP123423234IN"   // Read from APi and store here
  sessionStorage.barcode2 = JSON.stringify(barcode2);

    const navigate = useNavigate();
    const dispatch = useDispatch();
    const [pincodeData1, setPincodeData1] = useState('');
    const [pincodeData2, setPincodeData2] = useState('');

    const [user, setUser] = React.useState({
        fname: '',
        lname: '',
        rfname: '',
        rlname: '',
        address1: '',
        pincode1: '',
        city1: '',
        state1:'',
        phone1:phone,
        address2: '',
        pincode2: '',
        city2: '',
        state2:'',
        phone2:'',
        weight: '',
        price: '',
        accesstoken: '',
        reference: '',
        postType: '',
        barcode:''

  });
  const handleChange = (e) => {
    setUser({
      ...user,
      [e.target.name]: e.target.value,
    });
  };

    const submitValue = (formData) => {
        console.log(formData)

    // dispatch(AddVehicleSlice(formData));
    // dispatch(getusers);
    // navigate('/');
  };
  const validateFields = (form) => {
    if (user.fname !== '' && user.lname !== '' ) {


        return true;
    }
    return false;
  };
  const handleSubmit = (e) => {
    client.close();
    e.preventDefault();
    if (validateFields(e.target)) {
      console.log(pincodeData1.pincode)
      var requestData ={
        user :{
            fname: user.fname,
            lname: user.lname,
            rfname: user.rfname,
            rlname: user.rlname,
            address1: user.address1,
            pincode1: pincodeData1.pincode,
            city1:  pincodeData1.city,
            state1: pincodeData1.stateName,
            phone1:user.phone1,
            address2: user.address2,
            pincode2:  pincodeData2.pincode,
            city2:  pincodeData2.city,
            state2:pincodeData2.stateName,
            phone2:user.phone2,
            weight: user.weight,
            price: user.price,
            accesstoken: user.accesstoken,
            reference: user.reference,
            postType:"Registered Post",
            barcode: barcode2
        },
      };
    //   submitValue(requestData);


      sessionStorage.user = JSON.stringify(requestData);
      sessionStorage.price = JSON.stringify(requestData.user.price);


        navigate('/PrintComponent');
    }
    else {
      alert('Please fill all the fields');
    }
  };

  return (
    <>
      <div className="header" id="myHeader">
        <div className='header-items'><h2>RegisteredPost</h2></div>

      </div>


      <div className="wrapper">


      <div className="row">
          <form onSubmit={(e) => handleSubmit(e)} className='add-vehicle-form'>

        <div className="input-field">
            <label>Weight(Automatic)</label>
            <input autoComplete="off" className="form-control" value={weight}
              name='weight'
              onChange={handleChange} />
                  </div>
          <div className="input-field">
            <label>Sender's Name</label>
            <input autoComplete="off"type="text" className="form-control" id="formGroupExampleInput" required
              name='fname' placeholder='first name'
              onChange={handleChange}
                      /><br />
                      <input autoComplete="off"type="text" className="form-control" id="formGroupExampleInput" required
              name='lname' placeholder='last name'
              onChange={handleChange}
            />
          </div>
          <div className="input-field">
            <label>Sender Address</label>
            <input autoComplete="off"type="text" className="form-control" id="formGroupExampleInput2" required
              name='address1'
              onChange={handleChange} />
            </div>
            <div className="input-field">
            <label>Sender's Mobile</label>
            <input autoComplete="off" className="form-control" value={phone}
              name='phone1'
              onChange={handleChange} />
                  </div>
          <div className="input-field">
            <label>Sender's PinCode</label>
            <Pincode
        invalidError="Please check pincode"
        lengthError="check length"
        getData={(data) => setPincodeData1(data)} name="pincode1"
      />

          </div>
<div>
          <label>Reciever's Name</label>
            <input autoComplete="off"type="text" className="form-control" id="formGroupExampleInput" required
              name='rfname' placeholder='first name'
              onChange={handleChange}
                      /><br />
                      <input autoComplete="off"type="text" className="form-control" id="formGroupExampleInput" required
              name='rlname' placeholder='last name'
              onChange={handleChange}
            />
          </div>
          <div className="input-field">
            <label>Reciever Address</label>
            <input autoComplete="off"type="text" className="form-control" id="formGroupExampleInput2" required
              name='address2'
              onChange={handleChange} />
                  </div>
                  <div className="input-field">
            <label>Reciever's PinCode</label>
            <Pincode
        invalidError="Please check pincode"
        lengthError="check length"
        getData={(data) => setPincodeData2(data)} name="pincode2"
      />

                  </div>


                  <div className="input-field">
            <label>Reciever's Mobile</label>
            <input autoComplete="off"type="text" className="form-control" id="formGroupExampleInput2" required
              name='phone2'
              onChange={handleChange} />
          </div>
          <div className="input-field">
            <label>Price(Automatic)</label>
            <input autoComplete="off"type="text" className="form-control" id="formGroupExampleInput2"
              name='price'

              onChange={handleChange} />
          </div>
                  <div >

            <button className='main-button' type="submit" >Register</button>

          </div>
        </form>

      </div>
      </div>
      </>
  );
};

