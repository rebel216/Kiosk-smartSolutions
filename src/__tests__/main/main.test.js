import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Main from '../../components/Main/Main';
import { render, screen } from '@testing-library/react';


test('Main component match snapshot, landing page', () => {
    const component = renderer.create(
        <BrowserRouter>
            <Main />
        </BrowserRouter>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Has a button that takes you to registration page', () => {
    render(
        <BrowserRouter>
            <Main />
        </BrowserRouter>
    );
    expect(screen.getByText(/Register/i).href).toMatch(/signup/i);
});
