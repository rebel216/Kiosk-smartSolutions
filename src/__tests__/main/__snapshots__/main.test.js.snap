// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component match snapshot, landing page 1`] = `
<main
  className="splashscreen-container"
>
  <section
    className="splashscreen-content"
  >
    <h2
      className="splashscreen-message"
    >
      Welcome To Moblity Future
    </h2>
    <a
      className="button splashscreen-button"
      href="/signup"
      onClick={[Function]}
    >
      Register
    </a>
  </section>
</main>
`;
